eclair-core/src/test/scala/fr/acinq/eclair/channel/states/e/OfflineStateSpec.scala

```diff
@@ -534,4 +534,69 @@ class OfflineStateSpec extends TestkitBaseClass with StateTestsHelperMethods {
     sender.send(bob, CurrentFeerates(highFeerate))
     bob2blockchain.expectMsg(PublishAsap(bobCommitTx))
   }
+
+  test("re-send channel_update at reconnection for private channels") { f =>
+    import f._
+    val sender = TestProbe()
+
+    // we simulate a disconnection / reconnection
+    sender.send(alice, INPUT_DISCONNECTED)
+    sender.send(bob, INPUT_DISCONNECTED)
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+    sender.send(alice, INPUT_RECONNECTED(alice2bob.ref, aliceInit, bobInit))
+    sender.send(bob, INPUT_RECONNECTED(bob2alice.ref, bobInit, aliceInit))
+    alice2bob.expectMsgType[ChannelReestablish]
+    bob2alice.expectMsgType[ChannelReestablish]
+    bob2alice.forward(alice)
+    alice2bob.forward(bob)
+
+    // at this point the channel isn't deeply buried: channel_update isn't sent again
+    alice2bob.expectMsgType[FundingLocked]
+    bob2alice.expectMsgType[FundingLocked]
+    alice2bob.expectNoMsg()
+    bob2alice.expectNoMsg()
+
+    // we make the peers exchange a few messages
+    addHtlc(250000000 msat, alice, bob, alice2bob, bob2alice)
+    crossSign(alice, bob, alice2bob, bob2alice)
+
+    // we simulate a disconnection / reconnection
+    sender.send(alice, INPUT_DISCONNECTED)
+    sender.send(bob, INPUT_DISCONNECTED)
+    awaitCond(alice.stateName == OFFLINE)
+    awaitCond(bob.stateName == OFFLINE)
+    sender.send(alice, INPUT_RECONNECTED(alice2bob.ref, aliceInit, bobInit))
+    sender.send(bob, INPUT_RECONNECTED(bob2alice.ref, bobInit, aliceInit))
+    alice2bob.expectMsgType[ChannelReestablish]
+    bob2alice.expectMsgType[ChannelReestablish]
+    bob2alice.forward(alice)
+    alice2bob.forward(bob)
+
+    //  point the channel still isn't deeply buried: channel_update isn't sent again
+    alice2bob.expectNoMsg()
+    bob2alice.expectNoMsg()
+
+    // funding tx gets 6 confirmations, channel is private so there is no announcement sigs
```

Comment:
You should have an `alice2bob.expectNoMsg()` to verify that no announcement sig is sent, don't you?

Reply:
We receive `ChannelUpdate` instead.

Reply:
Isn't there a way to expect no message of type X?